engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala

```diff
@@ -53,7 +53,7 @@ class UpsertVisualizationJob(
   override def equalsTo(that: UniqueJob[_]): Boolean =
     that match {
       case that: UpsertVisualizationJob =>
-        this.expressionId == that.expressionId
+        this.expressionId == that.expressionId && this.visualizationId == that.visualizationId
```

Comment:
This was the root of the problem.

Reply:
Might deserve a simple unit test comparing two job instances for equality. The `RuntimeVisualizationsTest` may contain such check, but it is very hidden in the whole logic of the test.

Reply:
Added

Reply:
I see 1d8cd25e9315e3348d8700778105001903eb3b8d, thanks.